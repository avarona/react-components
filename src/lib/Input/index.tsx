import * as React from "react";
import classnames from "classnames";
import styles from "./styles.module.scss";
import Icon from "../Icon/index";

type Props = {
  addClass?: string;
  defaultValue?: string;
  onChange?: () => void;
  onKeyPress?: (value: string) => void;
  width?: number;
  leftIcon?: string;
  rightIcon?: string;
  disabled?: boolean;
  placeholder?: string;
};

const Input = ({
  addClass,
  defaultValue,
  onChange,
  width,
  leftIcon,
  rightIcon,
  disabled,
  onKeyPress,
  placeholder
}: Props) => {
  const [text, setText] = React.useState(defaultValue || "");

  return (
    <div
      style={{ width }}
      className={classnames(addClass, styles.input, {
        [styles.disabled]: disabled
      })}
    >
      {leftIcon && (
        <Icon
          addClass={classnames(styles.icon, styles.left)}
          img={leftIcon}
          size={20}
        />
      )}

      {rightIcon && (
        <Icon
          addClass={classnames(styles.icon, styles.right)}
          img={rightIcon}
          size={20}
        />
      )}

      <input
        type='text'
        value={text}
        disabled={disabled}
        placeholder={placeholder}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          setText(e.currentTarget.value);
          onChange && onChange();
        }}
        onKeyPress={(e: React.KeyboardEvent) => {
          if (e.key === "Enter") {
            onKeyPress && onKeyPress(text);
            setText("");
          }
        }}
      />
    </div>
  );
};

export default Input;
