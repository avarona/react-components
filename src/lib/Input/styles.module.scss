.input {
  display: flex;
  position: relative;
  align-items: center;

  input[type="text"] {
    outline: none;
    height: 40px;
    width: 100%;
    padding: 0 4px 0 8px;
    border: #d9d9d6 1px solid;
    border-radius: 4px;
    font-size: 14px;
  }
}

.icon {
  position: absolute;
  cursor: text;

  &.left {
    left: 8px;

    & + input[type="text"] {
      padding-left: 32px;
    }
  }

  &.right {
    right: 8px;

    & + input[type="text"] {
      padding-right: 32px;
    }
  }
}

.disabled {
  input[type="text"] {
    border-color: #aaa;
    cursor: not-allowed;
  }
}
