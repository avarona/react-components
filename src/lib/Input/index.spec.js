import React from "react";
import { shallow } from "enzyme";

import Input from "./";

describe("Input Component", () => {
  it("renders", () => {
    const wrapper = shallow(<Input />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("input").exists()).toBe(true);
  });
});
