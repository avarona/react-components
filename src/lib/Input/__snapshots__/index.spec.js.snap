// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders 1`] = `"<div class=\\"input\\"><input type=\\"text\\" value=\\"\\"/></div>"`;
